packages/developer-portal/src/components/pages/v8/pipeline/index.tsx

```diff
@@ -1,12 +1,49 @@
 import { PipelineModelInterface } from '@reapit/foundations-ts-definitions'
 import { useReapitGet } from '@reapit/utils-react'
-import React from 'react'
+import React, { useState } from 'react'
 import { CreatePipeline } from './create-pipeline'
 import { PipelineDeploymentInfo } from './deployment-info'
 import { reapitConnectBrowserSession } from '../../../../core/connect-session'
 import { GetActionNames, getActions } from '@reapit/utils-common'
 import { BodyText, FlexContainer, Loader } from '@reapit/elements'
-import { useReapitConnect } from '@reapit/connect-session'
+import { ReapitConnectSession, useReapitConnect } from '@reapit/connect-session'
+import { PusherProvider, useEvent, useChannel } from '@harelpls/use-pusher'
+import { URLS, COGNITO_HEADERS } from '../../../../constants/api'
+
+const PipelineWrapper = ({
```

Comment:
And again for these two components, can you break out the logic, break into separate components, add tests and props types please